### PR TITLE
[SYCL] Add a test with rebound usm_allocator

### PR DESCRIPTION
**Pull request by aelovikov-intel**

### CODE REVIEW

**Review by gmlueck — COMMENTED** [review]
LGTM

**Review by smaslov-intel — DISMISSED** [review]

File: SYCL/USM/allocator_rebind.cpp [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+bool is_overaligned(Allocator alloc, size_t Align,
+                    int Line = __builtin_LINE()) {
+  decltype(alloc.allocate(1)) Ptrs[10];
+  for (auto *&Elem : Ptrs)
+    Elem = alloc.allocate(1);
+
+  int NumExactlyAligned = 0;
+
+  for (auto *Ptr : Ptrs) {
+    auto Val = reinterpret_cast<uintptr_t>(Ptr);
```
Comment (cperkinsintel) [inline SYCL/USM/allocator_rebind.cpp]: maybe rename `Val` to `Addr`  ? that might be clearer

Comment (aelovikov-intel) [reply SYCL/USM/allocator_rebind.cpp]: I didn't want to do that because its type wouldn't be a pointer. Can rename if you insist though.

Comment (cperkinsintel) [reply SYCL/USM/allocator_rebind.cpp]: I wouldn't insist, but maybe persuade?  
If someone isn't reading this closely, then they'll likely conclude that `Val` is just the value held by the pointer. But clearly, we are testing the actual address of the pointer to ensure it's aligned. So I think `Addr`, or similar,  is clearer for the upcoming operations. Also, it makes people pay attention.  "Wow, he wants the address? That is so cool, I'd like to meet this guy"

**Review by cperkinsintel — APPROVED** [review]
LGTM

**Pull request merged**